src/main/java/com/minecolonies/core/items/MineColoniesArmorMaterial.java

```diff
@@ -1,92 +1,96 @@
+    });
 
-    private final String     name;
-    private final int        maxDamageFactor;
-    private final int[]      damageReductionAmountArray;
-    private final int        enchantability;
-    private final SoundEvent soundEvent;
-    private final float      toughness;
-    private final Ingredient repairMaterial;
+    private final String                      name;
+    private final int                         durabilityMultiplier;
+    private final Map<Type, Integer>          protectionFunctionForType;
+    private final int                         enchantmentValue;
+    private final SoundEvent                  sound;
+    private final float                       toughness;
+    private final float                       knockbackResistance;
+    private final LazyLoadedValue<Ingredient> repairIngredient;
 
     public MineColoniesArmorMaterial(
-      @NotNull final String name,
-      final int maxDamageFactor,
-      final int[] damageReductionAmountArray,
-      final int enchantability,
-      @NotNull final SoundEvent soundEvent,
+      final String name,
+      final int durabilityMultiplier,
+      final Map<Type, Integer> protectionFunctionForType,
+      final int enchantmentValue,
+      final SoundEvent sound,
       final float toughness,
-      @NotNull final Ingredient repairMaterial)
+      final float knockbackResistance,
+      final Supplier<Ingredient> repairIngredient)
     {
         this.name = name;
-        this.maxDamageFactor = maxDamageFactor;
-        this.damageReductionAmountArray = damageReductionAmountArray;
-        this.enchantability = enchantability;
-        this.soundEvent = soundEvent;
+        this.durabilityMultiplier = durabilityMultiplier;
+        this.protectionFunctionForType = protectionFunctionForType;
+        this.enchantmentValue = enchantmentValue;
+        this.sound = sound;
         this.toughness = toughness;
-        this.repairMaterial = repairMaterial;
+        this.knockbackResistance = knockbackResistance;
+        this.repairIngredient = new LazyLoadedValue<>(repairIngredient);
     }
 
-    @Override
-    public int getDurabilityForType(final ArmorItem.Type equipmentSlotType)
+    public int getDurabilityForType(@NotNull ArmorItem.Type type)
     {
-        return MAX_DAMAGE_ARRAY[equipmentSlotType.ordinal()] * this.maxDamageFactor;
+        return HEALTH_FUNCTION_FOR_TYPE.get(type) * this.durabilityMultiplier;
     }
 
-    @Override
-    public int getDefenseForType(final ArmorItem.Type p_267168_)
+    public int getDefenseForType(@NotNull ArmorItem.Type type)
     {
-        return this.damageReductionAmountArray[p_267168_.ordinal()];
+        return this.protectionFunctionForType.get(type);
     }
 
-    @Override
     public int getEnchantmentValue()
     {
-        return enchantability;
+        return this.enchantmentValue;
     }
 
-    @Override
-    public SoundEvent getEquipSound()
+    public @NotNull SoundEvent getEquipSound()
```

Comment:
while this is technically more correct, i think we put annotations on separate lines

Reply:
Yeah, idea puts it there even though our style is above, idk why